### PR TITLE
[Location Share] - Expanded map state when no more live location shares (PSG-629)

### LINKED ISSUE #6635: [Location Share] - Expanded map state when no more live location shares
#### What would you like to do?
When I am in the expanded map view and all user’s live-shares have ended:

- The pin disappears but the map remains at the position and zoom level that I had it
- I can still interact with the map.
- The share icon at the top-right is removed
- The bottom tray shows a greyed-out share icon and ‘live location ended’.
- This is only available if I am in the expanded map-view at the moment that the share ends. I cannot re-enter this view by clicking on the tile for a stopped live-share.

#### Why would you like to do it?
To improve UX.

#### How would you like to achieve it?
See https://www.figma.com/file/uthEK6xuo9hSQRdWvPAre5/Location-Sharing?node-id=402%3A50710 

### Have you considered any alternatives?

_No response_

### Additional context

_No response_

### Are you willing to provide a PR?

Yes

### PR DESCRIPTION
**Pull request by mnaturel**
## Type of change

- [ ] Feature
- [ ] Bugfix
- [ ] Technical
- [x] Other : Enhancement

## Content

Adding a dedicated view when there are no more active lives when in the expanded map screen.

## Motivation and context

Closes #6635 

## Tests

- Start a live location share in a room
- Go to the expanded map view when pressing the corresponding tile on the timeline
- Check the users list of active lives is correctly displayed
- Stop the live
- Check the users list dissapears
- Check a banner saying the live is ended is displayed a the bottom

## Tested devices

- [ ] Physical
- [x] Emulator
- OS version(s): Android 11

## Checklist

- [ ] Changes has been tested on an Android device or Android emulator with API 21
- [x] UI change has been tested on both light and dark themes
- [ ] Accessibility has been taken into account. See https://github.com/vector-im/element-android/blob/develop/CONTRIBUTING.md#accessibility
- [x] Pull request is based on the develop branch
- [x] Pull request includes a new file under ./changelog.d. See https://github.com/vector-im/element-android/blob/develop/CONTRIBUTING.md#changelog
- [x] Pull request includes screenshots or videos if containing UI changes
- [ ] Pull request includes a [sign off](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#sign-off)
- [x] You've made a self review of your PR
- [ ] If you have modified the screen flow, or added new screens to the application, you have updated the test [UiAllScreensSanityTest.allScreensTest()](https://github.com/vector-im/element-android/blob/main/vector/src/androidTest/java/im/vector/app/ui/UiAllScreensSanityTest.kt#L73)

### CODE REVIEW

File: vector/src/main/java/im/vector/app/features/home/room/detail/timeline/item/MessageLiveLocationItem.kt [hunk 1]
```diff
@@ -26,8 +26,8 @@ import im.vector.app.core.resources.toTimestamp
 import im.vector.app.core.utils.DimensionConverter
 import im.vector.app.features.home.room.detail.RoomDetailAction
 import im.vector.app.features.home.room.detail.timeline.style.TimelineMessageLayout
-import im.vector.app.features.location.live.LocationLiveMessageBannerView
 import im.vector.app.features.location.live.LocationLiveMessageBannerViewState
+import im.vector.app.features.location.live.LocationLiveRunningBannerView
 import org.threeten.bp.LocalDateTime
 
 @EpoxyModelClass
```
Comment (Florian14) [inline vector/src/main/java/im/vector/app/features/home/room/detail/timeline/item/MessageLiveLocationItem.kt:33]: Not a part of this PR but to be consistent, the class name should be `MessageLocationLiveItem` wdyt?

File: vector/src/main/java/im/vector/app/features/location/live/LocationLiveEndedBannerView.kt [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+                0,
+                0
+        ).run {
+            try {
+                setBackgroundAlpha(this)
+                setIconMarginStart(this)
+            } finally {
+                recycle()
+            }
+        }
```
Comment (Florian14) [inline vector/src/main/java/im/vector/app/features/location/live/LocationLiveEndedBannerView.kt:52]: I think you can simplify like this
```suggestion
        context.obtainStyledAttributes(
                attrs,
                R.styleable.LocationLiveEndedBannerView,
                0,
                0
        ).use {
                setBackgroundAlpha(it)
                setIconMarginStart(it)
        }
```

File: vector/src/main/java/im/vector/app/features/location/live/LocationLiveEndedBannerView.kt [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+    }
+
+    private fun setBackgroundAlpha(typedArray: TypedArray) {
+        val withAlpha = typedArray.getBoolean(R.styleable.LocationLiveEndedBannerView_locLiveEndedBkgWithAlpha, false)
+        binding.locationLiveEndedBannerBackground.alpha = if (withAlpha) 0.75f else 1f
```
Comment (Florian14) [inline vector/src/main/java/im/vector/app/features/location/live/LocationLiveEndedBannerView.kt:57]: Should the 0.75f alpha value be a const?

**Review by Florian14 — APPROVED** [review]
Not tested but changes look fine. Minor comments not blocking for the merge

File: vector/src/main/java/im/vector/app/features/home/room/detail/timeline/item/MessageLiveLocationItem.kt (hunk 1 above)
Comment (mnaturel) [reply vector/src/main/java/im/vector/app/features/home/room/detail/timeline/item/MessageLiveLocationItem.kt:33]: I think I will create a dedicated PR to align the naming of all components. I think I will keep the order `LiveLocation`.

File: vector/src/main/java/im/vector/app/features/location/live/LocationLiveEndedBannerView.kt (hunk 2 above)
Comment (mnaturel) [reply vector/src/main/java/im/vector/app/features/location/live/LocationLiveEndedBannerView.kt:52]: Yes! Thanks for the suggestion, I didn't see there were the extension `use` for `TypedArray`.

Comment (mnaturel) [reply vector/src/main/java/im/vector/app/features/location/live/LocationLiveEndedBannerView.kt]: Updated, see: [d62cf70](https://github.com/vector-im/element-android/pull/6636/commits/d62cf70b8c01496cc07acca12b57f41b2d7b9e1a)

File: vector/src/main/java/im/vector/app/features/location/live/LocationLiveEndedBannerView.kt (hunk 3 above)
Comment (mnaturel) [reply vector/src/main/java/im/vector/app/features/location/live/LocationLiveEndedBannerView.kt]: Yes we could have a const for better clarity. See [47960c0](https://github.com/vector-im/element-android/pull/6636/commits/47960c0dfd11c5bc93876aacbe4942f91e261d78)

File: vector/src/main/java/im/vector/app/features/location/live/LocationLiveEndedBannerView.kt (hunk 2 above)
Comment (Florian14) [reply vector/src/main/java/im/vector/app/features/location/live/LocationLiveEndedBannerView.kt]: Not an issue but you missed the shortcut `context.theme.obtainStyledAttributes` -> `context.obtainStyledAttributes`

Comment (mnaturel) [reply vector/src/main/java/im/vector/app/features/location/live/LocationLiveEndedBannerView.kt]: Yes indeed, thanks. Updated here: [0750e80](https://github.com/vector-im/element-android/pull/6636/commits/0750e80f04918b9a0e4649193a69be805cd032c9)

**Review by Florian14 — APPROVED** [review]

**Pull request merged**